www/src/components/Search/Search.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {Component} from "react"
+import {Link} from "gatsby"
+import {Index} from "elasticlunr"
+import {css} from "@emotion/core"
+import {ISearchProps, ISearchState} from "./ISearch"
+import {searchLabelStyle, searchBoxStyle} from "./Search.css"
+
+//TODO: Figure out how to give type to "this.index" from constructor.
+export default class Search extends Component<ISearchProps, ISearchState> {
+	public constructor(props) {
+		super(props)
+		this.state = {
+			query: ``,
+			results: [],
+		}
+	}
+
+	private getOrCreateIndex = () => (this.index ? this.index : Index.load(this.props.searchIndex)) // Create an elastic lunr index and hydrate with graphql query results
```

Comment:
Needs an underscore prefix `_ getOrCreateIndex`

Could also be renamed to just `_getIndex`

Reply:
yep will do!